Assets/Scripts/PlayerCharacter.cs

```diff
@@ -0,0 +1,256 @@
+    void Start() {
+        audioSource = GetComponent<AudioSource>();
+        jumpNumber = 0;
+        jumpCooldown = 0;
+        anim.SetBool("IsDead", false); //starting game not dead
+    }
+
+    // Update is called once per frame
+    void Update() {
+        if (Time.realtimeSinceStartup - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            if (Input.GetButtonDown("Jump") && jumpCooldown == 0)
+            {
+                Jump();//allows player to jump and checks to make sure character can't jump when dead
+            }
+            deathTimer = 0;
+        }
+        if (isInDeath)
+        {
+            anim.SetBool("IsDead", true);
+            if (Time.realtimeSinceStartup - deathTimer >= timeToDeath)
+            {
+                Respawn();//if death timer has gone up and matches time to death, it will enter the next stage of respawn.
+                GetComponent<Rigidbody2D>().velocity = new Vector2(0, 0);
+            }
+            GetComponent<Rigidbody2D>().velocity = myRigidBody.velocity/deathVector;
+        }
+        GetMovementInput();
+        if (jumpNumber > 0)
+        {
+            speedJumpReducer = 2; // reduces jump height with each subsequent jump.
+        }
+        /*if (Input.GetKey(KeyCode.Space) && jumpCooldown == 0)
+        {
+            Jump();
+        }*/
+        if (jumpCooldown > 0)
+        {
+            jumpCooldown--;
+        }
+        if (myRigidBody.velocity.y == 0)
+        {
+            jumpNumber = 0; //if the character is not moving vertically, the player has taken at least a moment to rest before jumping again.
+        }
+        if (jumpNumber == 0)
+        {
+            speedJumpReducer = 1; //first jump has no vertical distance reducer.
+        }
+        UpdateAnimationParameters();
+
+    }
+
+    private void UpdateAnimationParameters()
+    {
+        grounded = Physics2D.OverlapCircle(groundCheck.position, groundRadius, whatIsGround);
+        anim.SetBool("IsOnGround", grounded);
+        //sets tells animator if character is falling or on the ground.
+        if (myRigidBody.velocity.y < 0 && grounded == false)
+        {
+            falling = true;
+        }
+        else if (grounded==true || myRigidBody.velocity.y <= 0)
+        {
+            falling = false;
+        }
+        anim.SetBool("Falling", falling);
+        anim.SetFloat("Speed", Math.Abs(myRigidBody.velocity.x)); //lets the animator know the speed of player.
+    }
+
+    private void FixedUpdate()
+    {
+        UpdatePhysicsMaterial();
+        if (Time.time - deathTimer >= timeToDeath + 0.5f || deathTimer == 0)
+        {
+            //character cannot use the move function when dying.
+            if (Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.A))
+            {
+                Move();//lets player move a fixed amount not dependent on frames
+            }
+            //player flips sprite when going opposite direction of prior movement.
```

Comment:
You could probably extract this into a `FlipIfNeeded` method (or pick your own name that makes sense) with a summary.